### ISSUE TITLE
Can't make the code it work

### ISSUE THREAD

**Issue by dleunghk** [cristianbuse/Excel-VBA-SafeTimers#1]
Thanks for creating the safe API timer, I have been hit by the sudden crash hundreds of times in the past.

I tried both methods, but unfortunately both of them didn't seem to work.  I used the latest Office 365 64-bit on Windows 10 22H2 64-bit.  I tried uninstalling Office 365 and do a fresh installation, still didn't work.

For the API method, I can see that the new Excel instance and file was created in the task manager, but when the code executed to the end of InitTimer function and returned to SetTimer function, the new Excel instance was killed and therefore no timer can be initialised by the next few lines of code. I checked the temp directory and found the Excel file was properly initialised and saved.

For the native method, once the Demo code is run, the Excel just stuck in infinite loop.

Is there any thing or dependency I need to setup before either method works? Thank you in advance for your advice.

**Comment by cristianbuse** [comment]
Hi @dleunghk ,

There should be nothing to setup (no dependency). It should work directly. Well, it worked for all of the computers I tested on. It might be that something is specific to your configuration or maybe an antivirus is in the way.

Have you downloaded the [Demo book](https://github.com/cristianbuse/Excel-VBA-SafeTimers/raw/master/ExcelVBA%20SafeTimers_Solution%201%20-%20Native%20(Win%20%2B%20Mac)_Demo.xlsm) and ran the demo or did you import the modules into a bigger project? Just trying to isolate where the problem is.

**Comment by cristianbuse** [comment]
Forgot to ask, where do you use the timers? What do you need them for? I ask because I might suggest alternatives based on the specific need.

**Comment by dleunghk** [comment]
Dear Cristian,
Thanks for the advice, I downloaded both Demo books, and it seems that the native one worked, while the API one still didn't.  I may need some time to compare with the two files I created through .bas and .cls import.
Actually I have an Excel which streams real time stock data and I need a timer to run every 20 seconds to do some calculations on the data.
I am currently using the API callback which crash Excel often whenever I edited the VBA code when the timer was running.  And whenever I debug and break any VBA code, the timer cannot be stopped by KillTimer.   
Will let you know why my files didn't work.  Thanks again for your help and your excellent safe timer codes.

Rgds,Darik    Cristian Buse ***@***.***>) 在 2022年12月12日 星期一 GMT 下午12:19:38 寫道：  
 
 


Hi @dleunghk ,

There should be nothing to setup (no dependency). It should work directly. Well, it worked for all of the computers I tested on. It might be that something is specific to your configuration or maybe an antivirus is in the way.

Have you downloaded the Demo book and ran the demo or did you import the modules into a bigger project? Just trying to isolate where the problem is.
